src/extension/commandPaletteHandler.ts

```diff
@@ -96,7 +96,8 @@ export class CommandPaletteHandler {
 
         return this.reactNativePackager.start()
             .then(() => {
-                return new CommandExecutor(this.workspaceRoot).spawnReactCommand(command, args, null);
+                return new CommandExecutor(this.workspaceRoot).spawnReactCommand(command,
+                    args, /*waitForExit*/true, /*options*/null).outcome;
```

Comment:
`outcome` is already a promise, you can merge this section with the next section.